swap/src/protocol/alice/event_loop.rs

```diff
@@ -228,24 +227,24 @@ impl EventLoop {
                         }
                     }
                 },
-                swap_response = self.send_swap_response.next().fuse() => {
+                swap_response = self.send_swap_response.recv().fuse() => {
                     if let Some((channel, swap_response)) = swap_response  {
-                        self.swarm.send_swap_response(channel, swap_response);
+                        self.swarm.send_swap_response(channel, swap_response).expect("Could not send amounts message");
```

Comment:
I doubt that panicking here is the desired long-term strategy. That will tear-down the whole event loop.

Reply:
Even if tear-down is desired, you probably just want to return an error from the whole function :)

Reply:
This code is executed in a loop inside the event_loop.run() function. We spawn it into an async task. How does this interact with returning an error? Does it stop looping executing and return the error?

Reply:
Yes, it will break the loop and end the task. If you are `await`ing the `JoinHandle` that you get returned from `spawn`ing the task somewhere, then that will resolve to the `Err`.

Reply:
Not sure if we discussed that after or before this change but best for now to log any error you encounter and let the loop run.